src/sonic.c

```diff
@@ -319,6 +319,28 @@ XML_parser_general(void *data, const char *elem, const char **attr)
     LinkTable_add(linktbl, link);
 }
 
+static void sanitize_LinkTable(LinkTable *linktbl) {
```

Comment:
Could you please explain what caused the problem, and why this works?

Reply:
Some special characters can be used in elements name of a subsonic server. This is the case on my Funkwhale instance for example. If the name is a single dot, this has already a special meaning and will not be accessible. I guess the IO error is especially thrown by other programs when the name contains a `/` since it is the path separator in UNIX systems. So we have the choice to rename these elements to make specials characters disappear or to discard these elements to avoid problems with their names.